roles/docker/tasks/main.yml

```diff
@@ -22,4 +22,6 @@
 
 - name: restart docker
   shell: systemctl daemon-reload && systemctl stop docker && systemctl start docker-tcp.socket && systemctl start docker
-  when: docker_env_conf_result.changed or docker_tcp_conf_result.changed or docker_svc_conf_result.changed
+  when: docker_install_result.changed or swarm_install_result.changed or docker_env_conf_result.changed or
```

Comment:
well `base` is meant to be run just once (when the os image is built)..so `docker` role is run on an image configured from `base` but it might not be in same host-group in which case this will fail.

I think `base` role itself needs to be cleaned up to not install docker and move it to `docker` role but unless you are taking care of that this will break existing suff.